app/components/message_attachments/attachment_footer.js

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import {Image, Text, View, Platform} from 'react-native';
+import PropTypes from 'prop-types';
+import truncate from 'lodash/truncate';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {MAX_ATTACHMENT_FOOTER_LENGTH} from 'app/constants/attachment';
+
+export default class AttachmentFooter extends PureComponent {
+    static propTypes = {
+        text: PropTypes.string,
+        icon: PropTypes.string,
+        theme: PropTypes.object.isRequired,
+    };
+
+    render() {
+        const {
+            text,
+            icon,
+            theme,
+        } = this.props;
+
+        if (!text) {
+            return null;
+        }
+
+        const style = getStyleSheet(theme);
+
+        return (
+            <View style={style.container}>
+                {Boolean(icon) &&
+                <Image
+                    source={{uri: icon}}
+                    key='footer_icon'
+                    style={style.icon}
+                />
+                }
+                {Boolean(text) &&
+                <Text
+                    key='footer_text'
+                    style={style.text}
+                >
+                    {truncate(text, {length: MAX_ATTACHMENT_FOOTER_LENGTH, omission: '…'})}
+                </Text>
+                }
+            </View>
+        );
+    }
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return {
+        container: {
+            flex: 1,
+            flexDirection: 'row',
+            marginTop: 5,
+        },
+        icon: {
+            height: 12,
+            marginRight: 5,
+            marginTop: Platform.OS === 'ios' ? 1 : 2,
```

Comment:
For some reason, the footer icon renders slightly lower on iOS.  This platform-specific style keeps the layout visually similar between iOS and Android. (See screenshots in this PR's description.)

Reply:
```suggestion
            ...Platform.select({
                marginTop: {
                    ios: 1,
                    android: 2,
                },
            }
```

Reply:
@enahum Thanks.  I wasn't able to get your exact suggestion to work - I'm using this syntax instead:

```js
...Platform.select({
    ios: {
        marginTop: 1,
    },
    android: {
        marginTop: 2,
    },
}),
```

Let me know what you think 🙂